src/compiler/checker.ts

```diff
@@ -23117,11 +23118,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 break;
             case SyntaxKind.Parameter:
                 const param = declaration as ParameterDeclaration;
+                let keywordKind: SyntaxKind | undefined;
                 if (isIdentifier(param.name) &&
-                    (isCallSignatureDeclaration(param.parent) || isMethodSignature(param.parent) || isFunctionTypeNode(param.parent)) &&
-                    param.parent.parameters.indexOf(param) > -1 &&
-                    (resolveName(param, param.name.escapedText, SymbolFlags.Type, undefined, param.name.escapedText, /*isUse*/ true) ||
-                    param.name.originalKeywordKind && isTypeNodeKind(param.name.originalKeywordKind))) {
+                        (isCallSignatureDeclaration(param.parent) || isMethodSignature(param.parent) || isFunctionTypeNode(param.parent)) &&
+                        param.parent.parameters.indexOf(param) > -1 &&
+                        (resolveName(param, param.name.escapedText, SymbolFlags.Type, undefined, param.name.escapedText, /*isUse*/ true) ||
+                        (keywordKind = stringToToken(param.name.escapedText as string)) && isTypeNodeKind(keywordKind))) {

```

Comment:
nit: Not sure if I agree with the double indentation here. If the condition runs into the body, I'd just put the trailing `) {` of the condition on its own line:

```suggestion
                    (isCallSignatureDeclaration(param.parent) || isMethodSignature(param.parent) || isFunctionTypeNode(param.parent)) &&
                    param.parent.parameters.indexOf(param) > -1 &&
                    (resolveName(param, param.name.escapedText, SymbolFlags.Type, undefined, param.name.escapedText, /*isUse*/ true) ||
                    (keywordKind = stringToToken(param.name.escapedText as string)) && isTypeNodeKind(keywordKind))
                ) {
```

I don't know if we've ever expressed a stylistic preference for this though.